zookeeper-server/src/main/java/org/apache/zookeeper/ClientCnxnSocketNetty.java

```diff
@@ -82,7 +84,7 @@
 
     ClientCnxnSocketNetty(ZKClientConfig clientConfig) throws IOException {
         this.clientConfig = clientConfig;
-        eventLoopGroup = NettyUtils.newNioOrEpollEventLoopGroup();
+        eventLoopGroup = NettyUtils.newNioOrEpollEventLoopGroup(1 /* nThreads */);
```

Comment:
What about adding a comment and explain the reason of this magic number?

Reply:
Also I wonder if you maybe want to allow configure this... Seems like there is a config object passed around anyway.

Reply:
@normanmaurer I think it's overkill, the client only has 1 socket so why would we need more than 1 thread?

Reply:
Ah ok ... i thought you may create multiple connections

Reply:
Each connection would have its own `ClientCnxnSocketNetty` instance.

Reply:
got it... You could consider sharing the `EventLoopGroup` between those if this is feasible. But again this is just an optimisation and I am not sure if this is easily archiveable

Reply:
Probably not worth the trouble. I suspect that by far the common case in a client is connecting to a single ZK ensemble (per client process), so this optimization would not buy us much and would introduce additional complexity.

Reply:
@ivmaykov yeah sounds like this not justify the complexity